File: plus_one.cpp
class Solution {
public:
    vector<int> plusOne(vector<int> &digits)
    {
        // Only one thing to worry. What if the result is one digit longer than the given number?
        int carry = 1;
        int current_digit = 0;
        int n = digits.size();

        int BASE = 10;
        vector<int> result(n, 0);
        int i = 0;
        for (i = n - 1; i >= 0; i--)
        {
            current_digit = digits[i] + carry;
            carry = current_digit / BASE;
            result[i] = current_digit % BASE;
        }
        if (carry == 1)
        {
            result.insert(result.begin(), 1);
        }
        return result;
    }
};

File: Python solutions/plus_one.py
class Solution:
    # @param digits, a list of integer digits
    # @return a list of integer digits
    def plusOne(self, digits):
        result = []
        carry = 1
        base = 10
        for i in digits[::-1]:
            result.append((i+carry) % base)
            carry = (i + carry) / base
        if carry != 0:
            result.append(carry)
        return result[::-1]

# if __name__ == '__main__':
#     a = Solution()
#     print a.plusOne([0])